src/map/script.cpp

```diff
@@ -6671,15 +6671,15 @@ BUILDIN_FUNC(inarray)
 	if (not_server_variable(*name) && !script_rid2sd(sd))
 		return SCRIPT_CMD_FAILURE;
 
-	array_size = script_array_highest_key(st, sd, name, ref) - 1;
+	array_size = script_array_highest_key(st, sd, name, ref);
```

Comment:
Cam we make `array_size` unsigned? if we put it into a signed int, the `array_size >= SCRIPT_MAX_ARRAYSIZE` comparisons don't make sense.

let's make it const as well to prevent anything from changing it after the 0 check.